### PR TITLE
Fix multi decommissioning 

### PR DESCRIPTION
**Pull request by cscetbon**
| Q               | A
| --------------- | ---
| Bug fix?        | yes

### Why?

When decommissioning multiple nodes, only the first one or at least not all are decommissioning but just removed.

This PR also adds unit tests related to the decommission to ensure it doesn't happen with new changes

### CODE REVIEW

File: pkg/controller/cassandracluster/pod_operation.go [hunk 1]
```diff
@@ -290,46 +299,43 @@ func (rcc *ReconcileCassandraCluster) ensureDecommission(cc *api.CassandraCluste
...
-			return breakResyncLoop, fmt.Errorf("For Status Ongoing we should have a PodLastOperation Pods item")
-		}
+	case api.StatusFinalizing:
 		lastPod, err := rcc.GetPod(cc.Namespace, podLastOperation.Pods[0])
 		if err != nil {
-			if !apierrors.IsNotFound(err) {
-				return breakResyncLoop, fmt.Errorf("failed to get last cassandra's pods '%s': %v",
-					podLastOperation.Pods[0], err)
+			if apierrors.IsNotFound(err) {
+				return rcc.deletePodPVC(cc, dcName, rackName, status, lastPod)
```
Comment (erdrix) [inline pkg/controller/cassandracluster/pod_operation.go:316]: is that what we want to do ? 
Was there a reason that we didn't remove it before ?

File: pkg/controller/cassandracluster/decommission_test.go [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+	rcc.client.Update(context.TODO(), pdb)
+
+	return rcc, req
+}
+
+func registerJolokiaOperationModeResponder(host podName, op operationMode) {
+	fmt.Println(string(op))
+	httpmock.RegisterResponder("POST", JolokiaURL(host.FullName, jolokiaPort),
+		httpmock.NewStringResponder(200, fmt.Sprintf(`{"request":
+											{"mbean": "org.apache.cassandra.db:type=StorageService",
```
Comment (erdrix) [inline pkg/controller/cassandracluster/decommission_test.go:48]: indentation ?

File: pkg/controller/cassandracluster/pod.go [hunk 3]
```diff
@@ -41,8 +41,8 @@ var reEndingNumber = regexp.MustCompile("[0-9]+$")
 // PodContainersReady returns true if all container in the Pod are ready
 func PodContainersReady(pod *v1.Pod) bool {
 	if pod.Status.ContainerStatuses != nil && len(pod.Status.ContainerStatuses) > 0 {
-		for _, c := range pod.Status.ContainerStatuses {
-			if c.Ready == false {
+		for _, containerStatus := range pod.Status.ContainerStatuses {
+			if containerStatus.Ready == false {
```
[the remark this answers is not in the document]
Comment (cscetbon) [reply pkg/controller/cassandracluster/pod.go:45]: right, I didn't look into that code but just renamed the variable. Doing the change

File: pkg/controller/cassandracluster/decommission_test.go (hunk 2 above)
Comment (cscetbon) [reply pkg/controller/cassandracluster/decommission_test.go:48]: nope it's an inner call

File: pkg/controller/cassandracluster/pod_operation.go (hunk 1 above)
Comment (cscetbon) [reply pkg/controller/cassandracluster/pod_operation.go:316]: I just changed when it's done to fix the problem

**Pull request merged**